src/Pos/Util/JsonLog.hs

```diff
@@ -77,3 +89,30 @@ showHash = sformat hashHexF
 -- | Returns event of created 'Block'.
 jlAdoptedBlock :: SscHelpersClass ssc => Block ssc -> JLEvent
 jlAdoptedBlock = JLAdoptedBlock . showHash . headerHash
+
+data JsonLogConfig
+    = JsonLogDisabled
+    | JsonLogConfig (MVar Handle) (JLTimedEvent -> IO Bool)
+
+jsonLogConfigFromHandle :: MonadIO m => Handle -> m JsonLogConfig
+jsonLogConfigFromHandle h = do
+    v <- newMVar h
+    return $ JsonLogConfig v (\_ -> return True)
+
+jsonLogReal
+    :: (ToJSON a, Ether.MonadReader' JsonLogConfig m, Mockable Catch m,
+        MonadIO m, WithLogger m)
+    => a -> m ()
+jsonLogReal x = do
```

Comment:
Yep. Unfortunately, this code isn't exported as a function, so I had to copy it.